src/server/dfly_main.cc

```diff
@@ -319,6 +328,24 @@ bool RunEngine(ProactorPool* pool, AcceptServer* acceptor) {
     }
   }
 
+  std::uint16_t admin_port = GetFlag(FLAGS_admin_port);
+  if (admin_port != 0) {
+    const std::string& admin_bind = GetFlag(FLAGS_admin_bind);
+    // Note passing the result of c_str() for empty string in optimized mode don't work, we must
+    // explicitly set this to null in this case
+    const char* interface_addr = admin_bind.empty() ? nullptr : admin_bind.c_str();
+    Listener* admin_listener = new Listener{Protocol::REDIS, &service};
+    error_code ec = acceptor->AddListener(interface_addr, admin_port, admin_listener);
+    if (ec) {
+      LOG(ERROR) << "Failed to open [" << (!interface_addr ? "any" : interface_addr)
```

Comment:
pls factor out `(!interface_addr ? "any" : interface_addr)` into a variable to remove duplication.

Reply:
OK